### PR TITLE
#1790 Use mapperPrism.values.nullValuePropertyMappingStrategy when retrieving NullValuePropertyMappingStrategy

### LINKED ISSUE #1790: 1.3.0.Final Regression in nullValuePropertyMappingStrategy when inherited
On version 1.3.0.Final: 

When using `nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS` in a MapperConfig
and `nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE` on the Mapper,
the generated code is not what the docs is specifying:

Mapper:
```
@Mapper(
        config = MappingConfig.class,
        nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE
)
public interface BuggedMapper {


    CarDto toCarDto(CarModel carModel);


    void toExistingCar(@MappingTarget CarDto carDto, CarModel carModel);

}
```

MappingConfig:
```
@MapperConfig(
        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS
)
public class MappingConfig {

}
```

Generated BuggedMapperImpl: (comment is mine)
```
@Override
    public void toExistingCar(CarDto carDto, CarModel carModel) {
        if ( carModel == null ) {
            return;
        }

        if ( carModel.getBrandName() != null ) {
            carDto.setBrandName( carModel.getBrandName() );
        }
// this else branch should not be here
        else {
            carDto.setBrandName( null );
        }
    }
```

Adding ` nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS` to the Mapper fixes it, as well as reverting to mapstruct 1.3.0.Beta2

I have attached a sample project that reproduces the issue
[mapstructBugs.zip](https://github.com/mapstruct/mapstruct/files/3062936/mapstructBugs.zip)

### PR DESCRIPTION
**Pull request by filiphr**
Thanks to @saw303 for the good reproducal.

Fixes #1790

### CODE REVIEW

File: processor/src/test/java/org/mapstruct/ap/test/bugs/_1790/Issue1790Test.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+@WithClasses({
+    Issue1790Config.class,
+    Issue1790Mapper.class,
+    Source.class,
+    Target.class,
+})
+public class Issue1790Test {
+
+    @Test
+    public void name() {
```
Comment (Captain1653) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_1790/Issue1790Test.java]: I'm not sure. Maybe should we rename this test?

Comment (filiphr) [reply processor/src/test/java/org/mapstruct/ap/test/bugs/_1790/Issue1790Test.java]: You are right

Comment (filiphr) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_1790/Issue1790Test.java]:
```suggestion
    public void shouldProperlyApplyNullValuePropertyMappingStrategyWhenInheriting() {
```

**Comment by filiphr** [comment]
Thanks for the review @Captain1653

**Pull request merged**